### PR TITLE
feat: respect rule levels defined in config

### PR DESCRIPTION
**Pull request by BRKalow**
🎟️ [Asana Task]()

---

## Description

Respect the rule level defined in config when calling `context.report()`. This ensures we can properly discern between warnings and errors, and handle the result accordingly.

## PR Checklist 🚀

- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Write a useful description (above) to give reviewers appropriate context.
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203332374839504

### CODE REVIEW

File: packages/content-conformance/src/__tests__/rules.test.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ describe('loadRule', () => {
 
 describe('loadRules', () => {
   test('loads internal rules', async () => {
-    const rules = await loadRules({ 'content-no-h1': 'error' })
+    const rules = await loadRules({ ['content-no-h1' as string]: 'error' })
```
Comment (BRKalow) [inline packages/content-conformance/src/__tests__/rules.test.ts:35]: This was to address a weird type error, which was saying 'content-no-h1' didn't satisfy the key type of `Record<string, ...>` 🤷

File: packages/content-conformance/src/engine.ts [hunk 2]
```diff
@@ -67,13 +67,17 @@ export class ContentConformanceEngine {
...
       contentFiles: this.contentFiles,
       dataFiles: this.dataFiles,
       report(message: string, file?: ContentConformanceFile, node?: Node) {
         if (file) {
-          file.message(message, node, rule.id)
+          const fileMessage = file.message(message, node, rule.id)
+
+          if (rule.level === 'error') {
+            fileMessage.fatal = true
+          }
```
Comment (BRKalow) [inline packages/content-conformance/src/engine.ts:80]: The rest is handled by the vfile message logic / implementation ✨

**Review by thiskevinwang — COMMENTED** [review]
General Q: Is there a way to associate [this future message](https://github.com/hashicorp/web-platform-packages/pull/118/files#diff-1efb399d9be7cd6c9ac3565df449d7f006b49fb428dd362041652a353489c882R51) with a rule and/or a severity?

**Comment by BRKalow** [comment]
@thiskevinwang yeah. if you assign the return value of `this.message()` to a variable and set `.fatal = true`, it should treat it as an error.

**Pull request merged**